lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +200,47 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._seed_urls = config.get_seed_urls()
+        self._config = config
+        self.urls = []
 
+    # Make sure that find_articles only iterates over
+    # seed URLs and stores newly collected ones,
+    # while all the extraction is performed via protected
+    # _extract_url method.
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        url = article_bs.get('href')
+        # url = re.compile(r'(?<=href=")[\w./:]+(?=")')
+        if isinstance(url, str):
+            return url
+        return ''
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            res = make_request(seed_url, self._config)
+            html = BeautifulSoup(res.content, 'lxml')
+            # print(html.find_all('a', class_="news-list-card"))
+            for paragraph in html.find_all('a', {'class': "news-list-card"}):
+                if len(self.urls) >= self._config.get_num_articles():
```

Comment:
consider moving this check 
as of now, it looks like you'll have to make one extra request which takes additional time etc.